linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -69,91 +88,100 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "Failed to read IDP distributionPoint",
+		if !dpName.ReadASN1(&dpName, cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) {
+			return &lint.LintResult{
+				Status:  lint.Warn,
+				Details: "Failed to read IDP distributionPoint fullName",
+			}
 		}
-	}
 
-	if !dpName.ReadASN1(&dpName, cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "Failed to read IDP distributionPoint fullName",
+		uriBytes := make([]byte, 0)
+		if !dpName.ReadASN1Bytes(&uriBytes, cryptobyte_asn1.Tag(6).ContextSpecific()) {
+			return &lint.LintResult{
+				Status:  lint.Warn,
+				Details: "Failed to read IDP URI",
+			}
 		}
-	}
 
-	uriBytes := make([]byte, 0)
-	if !dpName.ReadASN1Bytes(&uriBytes, cryptobyte_asn1.Tag(6).ContextSpecific()) {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "Failed to read IDP URI",
+		uri, err := url.Parse(string(uriBytes))
+		if err != nil {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "Failed to parse IDP URI",
+			}
 		}
-	}
-
-	uri, err := url.Parse(string(uriBytes))
-	if err != nil {
-		return &lint.LintResult{
-			Status:  lint.Error,
-			Details: "Failed to parse IDP URI",
+		if uri.Scheme != "http" {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "IDP URI MUST use http scheme",
+			}
 		}
-	}
-
-	if uri.Scheme != "http" {
-		return &lint.LintResult{
-			Status:  lint.Error,
-			Details: "IDP URI MUST use http scheme",
+		if !dpName.Empty() {
+			return &lint.LintResult{
+				Status:  lint.Warn,
+				Details: "IDP should contain only one distributionPoint",
+			}
 		}
-	}
-
-	if !dpName.Empty() {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "IDP should contain only one distributionPoint",
+		if !idpv.ReadASN1Bytes(&onlyContainsUserCerts, cryptobyte_asn1.Tag(1).ContextSpecific()) {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "Failed to read IDP onlyContainsUserCerts",
+			}
 		}
-	}
-
-	// Ensure that OnlyContainsUserCerts is True. We have to read this boolean as
-	// a byte and ensure its value is 0xFF because cryptobyte.ReadASN1Boolean
-	// can't handle custom encoding rules like this field's [1] tag.
-	if !idpv.PeekASN1Tag(cryptobyte_asn1.Tag(1).ContextSpecific()) {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "IDP should contain onlyContainsUserCerts",
+		if len(onlyContainsUserCerts) != 1 || onlyContainsUserCerts[0] != 0xFF {
+			return &lint.LintResult{
+				Status:  lint.Error,
+				Details: "IDP should set onlyContainsUserCerts: TRUE",
+			}
 		}
 	}
 
-	onlyContainsUserCerts := make([]byte, 0)
-	if !idpv.ReadASN1Bytes(&onlyContainsUserCerts, cryptobyte_asn1.Tag(1).ContextSpecific()) {
-		return &lint.LintResult{
-			Status:  lint.Error,
-			Details: "Failed to read IDP onlyContainsUserCerts",
+	if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(2).ContextSpecific()) {
```

Comment:
Honestly I think this can just be an `else`, instead of its own separate `if` clause.

This lets you move the `onlyContainsUserCerts` and `onlyContainsCACerts` predeclarations inside their own halves of the if/else conditional, and lets you remove the "don't set both" check on line 173 (because that will be handled by the "no other fields are set" check on line 180).